packages/code-du-travail-frontend/src/conventions/texte/SummaryItem.js

```diff
@@ -1,22 +1,31 @@
 import React from "react";
 import PropTypes from "prop-types";
-import styled, { css } from "styled-components";
+import styled from "styled-components";
 import { theme } from "@cdt/ui/";
+import { TocListItem, TocList, TocLink } from "./toc";
 
-const SummaryItem = ({ data, children, type, level, visible }) => {
+const SummaryItem = ({ data, children, type, level, expanded }) => {
   const { titre, id } = data;
   return (
-    <Wrapper level={level} title={`${type} ${id}, level ${level}`}>
-      <Title visible={visible}>
-        <TitleLink href={`#${id}`}>{titre}</TitleLink>
-      </Title>
-      {children &&
-        children.map(child => (
-          <SummaryItem key={child.data.id} level={level + 1} {...child}>
-            {child.children}
-          </SummaryItem>
-        ))}
-    </Wrapper>
+    <TocListItem level={level} type={type} id={id}>
+      <TocLinkNested type={type} level={level} id={id}>
+        {titre}
+      </TocLinkNested>
+      {children && (
+        <TocList expanded={expanded}>
+          {children.map(child => (
+            <SummaryItem
+              expanded={expanded}
+              key={child.data.id}
+              level={level + 1}
+              {...child}
+            >
+              {child.children}
```

Comment:
I didn't realize before but I'm very confused by the `children`  props here
React Element have a prop called children too which contains children Reacts Element of a givent ReactElements.

Reply:
yes these are the same `children` I am referencing here, the React props. My understanding is that if you call a component with something inside the tag like :
```
<Title>
  <div>bonjour</div> 
  <span>coucou</div>
</Title>
```
in the Title component, the `children` prop will contain these two subcomponents, the div and the span.

in the SummaryItem, I'm calling the component recursively, so I'm doing something like :

```
const SummaryItem = ({ children }) => (
  <React.Fragment>
  {children.map(child => (
    <SummaryItem>{child.children}</SummaryItem>
  )}
  </React.Fragment>
);
```

is it weird ? I could also pass down the children with `<SummaryItem children={child.children} />` but I don't feel like it's any better

Reply:
btw i think you should use `React.Children.map` https://reactjs.org/docs/react-api.html#reactchildren

Reply:
nice I didn't know about that I'll try @revolunet

Reply:
actually I think that the main problem was that I was mixing React's `children` and my own AST-like tree structure's `children`

Reply:
https://github.com/SocialGouv/code-du-travail-numerique/pull/847/commits/c5a33f8e99e45d7fb75da0586c59be0d90dd3ef9 @revolunet I think it's much clearer (even for me 😬 )

Reply:
thanks @adipasquale for the renaming 🙏